### PR TITLE
Glass Maintenance station trait

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a station trait that makes all maintenance airlocks glass
## Why It's Good For The Game
Makes it harder for antags to hide bodies/runes/etc
## Changelog
:cl:
add: glass maint station trait
/:cl:

### CODE REVIEW

**Comment by etherware-novice** [comment]
how do i replace floor

**Comment by Time-Green** [comment]
While you're at it, make every object inside maintenance made of glass

**Comment by etherware-novice** [comment]
?? how would that evne work

**Comment by Time-Green** [comment]
> I think Time Green is just joking

It was an optional suggestion but I was not joking

File: code/datums/station_traits/positive_traits.dm [hunk 1]
```diff
@@ -333,3 +333,28 @@
...
+		"In an effort to promote brand transparancy, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
+		"We over-ordered glass during construction, so maintenance gets glass windows.")
+
+/datum/station_trait/on_round_start()
+	for(var/area/station/maintenance/A in GLOB.areas)
+		for(var/turf/in_area as anything in A.get_contained_turfs())
+			for(var/obj/machinery/door/airlock/D in in_area)
+				D.opacity = FALSE
+				D.glass = TRUE
+				D.update_icon(ALL, 0)
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:360]: You need to use descript variable names.

Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:360]: and named arguments

**Comment by Serxule** [comment]
> how do i replace floor

ask someone who knows mapping i guess

**Comment by Jolly-66** [comment]
> Make every airlock and floor tile on the map glass

How to shatter performance (I don't know if it'll do that)

**Comment by Mothblocks** [comment]
> Adds an extra difficulty for most antags to stay hidden, and makes it slightly easier for crew to locate bodies

Why is this good for the game?

**Comment by etherware-novice** [comment]
it forces antags to be more creative with placement bc you cant just leave your secret setup right outside engineering you have to make sure nobody can see you from the hallway

File: code/datums/station_traits/positive_traits.dm [hunk 2]
```diff
@@ -333,3 +333,28 @@
 		return
 	var/obj/item/organ/internal/cybernetic = new cybernetic_type()
 	cybernetic.Insert(spawned, special = TRUE, drop_if_replaced = FALSE)
+
+/datum/station_trait/glass_maint
+	name = "Glass Maintinance"
+	trait_type = STATION_TRAIT_POSITIVE
+	show_in_report = TRUE
+	weight = 5
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:341]: weight on this needs to be lower

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
@Fikou I don't really care about it being a station trait I just think the actual text they have written is dumb

**Comment by Overuseddart** [comment]
> > > Adds an extra difficulty for most antags to stay hidden, and makes it slightly easier for crew to locate bodies
> > 
> > 
> > Why is this good for the game?
> 
> station traits are meant to modify the gameplay with unique effects that affect how you play making people able to see through maint doors is an unique effect like that, making people have to be more careful with their maint shenanigans

This doesnt make anyone play different other than antags, crew has no real risk in this trait, every tile being a glass floor or airlock would have an effect though. (And probably crash the server)

**Comment by etherware-novice** [comment]
not every tile, just every airlock and maybe floor tile

File: code/datums/station_traits/positive_traits.dm [hunk 3]
```diff
@@ -333,3 +333,28 @@
...
+		"In an effort to comply with Space Law section [rand(1,99999)].4, we have outfitted maintenence with glass windows.",
+		"An anomaly caused all the maintenance airlocks to get glass windows.",
+		"An intern went through and added glass to all the maintenance windows in their spare time.",
+		"In an effort to promote brand transparancy, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
+		"We over-ordered glass during construction, so maintenance gets glass windows.")
+
+/datum/station_trait/on_round_start()
+	for(var/area/station/maintenance/A in GLOB.areas)
+		for(var/turf/in_area as anything in A.get_contained_turfs())
+			for(var/obj/machinery/door/airlock/D in in_area)
```
Comment (Mothblocks) [inline code/datums/station_traits/positive_traits.dm:357]: You should loop over GLOB.airlocks instead

**Comment by Blevruz** [comment]
> > > > Adds an extra difficulty for most antags to stay hidden, and makes it slightly easier for crew to locate bodies
> > > 
> > > 
> > > Why is this good for the game?
> > 
> > 
> > station traits are meant to modify the gameplay with unique effects that affect how you play making people able to see through maint doors is an unique effect like that, making people have to be more careful with their maint shenanigans
> 
> This doesnt make anyone play different other than antags, crew has no real risk in this trait, every tile being a glass floor or airlock would have an effect though. (And probably crash the server)

it is also a significant buff to the maintenance death star

File: code/datums/station_traits/positive_traits.dm [hunk 4]
```diff
@@ -333,3 +333,34 @@
...
+	report_message = pick("In an effort to reduce maintenance casualties, glass windows have been fitted in every door.",
+		"We had a spare engineer, so we had them put glass in all maintenance doors.",
+		"To commemorate our miners, we put glass in all the airlocks. Unfortunately, we ran out after completing maintenance.",
+		"In an effort to comply with Space Law section [rand(1,99999)].4, we have outfitted maintenence with glass windows.",
+		"An anomaly caused all the maintenance airlocks to get glass windows.",
+		"An intern went through and added glass to all the maintenance windows in their spare time.",
+		"In an effort to promote brand transparancy, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
+		"We over-ordered glass during construction, so maintenance gets glass windows.")
+
+/datum/station_trait/on_round_start()
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:354]: this code runs on every single station trait

Comment (Fikou) [reply code/datums/station_traits/positive_traits.dm:354]: also why are we running this on roundstart??? that's just a large delay

File: code/datums/station_traits/positive_traits.dm [hunk 5]
```diff
@@ -333,3 +333,34 @@
...
+	for(var/area/station/maintenance/maint_area in GLOB.the_station_areas)
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:355]: as anything

File: code/datums/station_traits/positive_traits.dm [hunk 6]
```diff
@@ -333,3 +333,34 @@
...
+		for(var/turf/maint_floor as anything in maint_area.get_contained_turfs())
+
+			if(istype(maint_floor, /turf/open/floor/plating/reinforced))  // keeps reinf status
+				maint_floor.ChangeTurf(/turf/open/floor/glass/reinforced)
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:359]: is stupid, these arent comparable

File: code/datums/station_traits/positive_traits.dm [hunk 7]
```diff
@@ -333,3 +333,34 @@
...
+/datum/station_trait/glass_maint/New()
+	. = ..()
+	report_message = pick("In an effort to reduce maintenance casualties, glass windows have been fitted in every door.",
+		"We had a spare engineer, so we had them put glass in all maintenance doors.",
+		"To commemorate our miners, we put glass in all the airlocks. Unfortunately, we ran out after completing maintenance.",
+		"In an effort to comply with Space Law section [rand(1,99999)].4, we have outfitted maintenence with glass windows.",
+		"An anomaly caused all the maintenance airlocks to get glass windows.",
+		"An intern went through and added glass to all the maintenance windows in their spare time.",
+		"In an effort to promote brand transparancy, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
+		"We over-ordered glass during construction, so maintenance gets glass windows.")
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:352]: this doesnt mention anything about the floors

File: code/datums/station_traits/positive_traits.dm [hunk 8]
```diff
@@ -333,3 +333,34 @@
 		return
 	var/obj/item/organ/internal/cybernetic = new cybernetic_type()
 	cybernetic.Insert(spawned, special = TRUE, drop_if_replaced = FALSE)
+
+/datum/station_trait/glass_maint
+	name = "Glass Maintenance"
+	trait_type = STATION_TRAIT_POSITIVE
+	show_in_report = TRUE
+	weight = 4
```
Comment (Fikou) [inline code/datums/station_traits/positive_traits.dm:341]: this is still too high

File: code/datums/station_traits/positive_traits.dm (hunk 5 above)
Comment (Time-Green) [reply code/datums/station_traits/positive_traits.dm:355]: Psure he wants only the maint areas

Comment (Fikou) [reply code/datums/station_traits/positive_traits.dm:355]: youre right

File: code/datums/station_traits/positive_traits.dm (hunk 4 above)
Comment (etherware-novice) [reply code/datums/station_traits/positive_traits.dm:354]: typo, and how should i have it set up then

Comment (Fikou) [reply code/datums/station_traits/positive_traits.dm:354]: on new probably

**Comment by YakumoChen** [comment]
Not sure what making the doors see-through adds to the game

I came into the PR thinking it would make all the Hull windows in maint to regular glass (making maint hilariously dangerous) instead of this

**Comment by etherware-novice** [comment]
hull windows?

**Comment by YakumoChen** [comment]
Hull, meaning the outer shell to the statuon, meaning all the windows to space

File: code/datums/station_traits/positive_traits.dm [hunk 9]
```diff
@@ -333,3 +333,29 @@
...
+/datum/station_trait/glass_maint/New()
+	. = ..()
+	report_message = pick("In an effort to reduce maintenance casualties, glass windows have been fitted in every door.",
+		"We had a spare engineer, so we had them put glass in all maintenance doors.",
+		"To commemorate our miners, we put glass in all the airlocks. Unfortunately, we ran out after completing maintenance.",
+		"In an effort to comply with Space Law section [rand(1,99999)].4, we have outfitted maintenence with glass windows.",
+		"An anomaly caused all the maintenance airlocks to get glass windows.",
+		"An intern went through and added glass to all the maintenance windows in their spare time.",
+		"In an effort to promote brand transparancy, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
+		"We over-ordered glass during construction, so maintenance gets glass windows.")
```
Comment (Mothblocks) [inline code/datums/station_traits/positive_traits.dm:352]:
```suggestion
	report_message = pick(
		"In an effort to reduce maintenance casualties, glass windows have been fitted in every door.",
		"We had a spare engineer, so we had them put glass in all maintenance doors.",
		"To commemorate our miners, we put glass in all the airlocks. Unfortunately, we ran out after completing maintenance.",
		"In an effort to comply with Space Law section [rand(1,99999)].4, we have outfitted maintenance with glass windows.",
		"An anomaly caused all the maintenance airlocks to get glass windows.",
		"An intern went through and added glass to all the maintenance windows in their spare time.",
		"In an effort to promote brand transparency, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
		"We over-ordered glass during construction, so maintenance gets glass windows.",
	)
```

Comment (Mothblocks) [reply code/datums/station_traits/positive_traits.dm:352]: fixes typos and also follows the multithreading rules

File: code/datums/station_traits/positive_traits.dm [hunk 10]
```diff
@@ -333,3 +333,29 @@
...
+
+	for(var/area/station/maintenance/maint_area in GLOB.the_station_areas)
+		for(var/turf/maint_floor as anything in maint_area.get_contained_turfs())
+			maint_floor.ChangeTurf(/turf/open/floor/glass)
+
+			for(var/obj/machinery/door/airlock/affected_airlock in maint_floor)
+				affected_airlock.opacity = FALSE
+				affected_airlock.glass = TRUE
+				affected_airlock.update_icon(ALL)
```
Comment (Mothblocks) [inline code/datums/station_traits/positive_traits.dm:361]:
```suggestion
				affected_airlock.update_appearance(UPDATE_ICON)
```

File: code/datums/station_traits/positive_traits.dm (hunk 9 above)
Comment (etherware-novice) [reply code/datums/station_traits/positive_traits.dm:352]: i tried that originally and it spat a compile error

Comment (Mothblocks) [reply code/datums/station_traits/positive_traits.dm:352]: try it again

**Comment by LemonInTheDark** [comment]
> While you're at it, make every object inside maintenance made of glass

This means maint is fullbright, which vibes wrong

**Comment by ChungusGamer666** [comment]
make the walls made out of glass too that'd be funny

**Comment by calliophis-maculiceps** [comment]
this isn't game changing but station traits shouldn't be either, now just let us stack low impact station traits and we're golden

**Comment by LemonInTheDark** [comment]
Doesn't this kill nightmares? or am I silly

**Comment by Mothblocks** [comment]
i am also curious on that now @etherware-novice

File: code/datums/station_traits/positive_traits.dm [hunk 11]
```diff
@@ -333,3 +333,38 @@
...
+		"To commemorate our miners, we put glass in all the airlocks. Unfortunately, we ran out after completing maintenance.",
+		"In an effort to comply with Space Law section [rand(1,99999)].4, we have outfitted maintenance with glass windows.",
+		"An anomaly caused all the maintenance airlocks to get glass windows.",
+		"An intern went through and added glass to all the maintenance windows in their spare time.",
+		"In an effort to promote brand transparency, we're adding glass to all maintenance doors effective immediately.",  // my proudest dad joke
+		"We over-ordered glass during construction, so maintenance gets glass windows.",
+	)
+
+/datum/station_trait/glass_maint/on_round_start()
+
```
Comment (Mothblocks) [inline code/datums/station_traits/positive_traits.dm:357]:
```suggestion
```

File: code/datums/station_traits/positive_traits.dm [hunk 12]
```diff
@@ -333,3 +333,38 @@
...
+	for(var/area/station/maintenance/maint_area in GLOB.areas)
+		for(var/turf/maint_floor as anything in maint_area.get_contained_turfs())
+
```
Comment (Mothblocks) [inline code/datums/station_traits/positive_traits.dm:360]:
```suggestion
```

File: code/datums/station_traits/positive_traits.dm [hunk 13]
```diff
@@ -333,3 +333,38 @@
...
+			if(istype(maint_floor, /turf/closed))  // replace walls with fulltile windows to not space everything
+				new /obj/structure/window/reinforced/fulltile(maint_floor)
```
Comment (Mothblocks) [inline code/datums/station_traits/positive_traits.dm:362]: Wow I definitely do not agree with this at all

**Review by Mothblocks — CHANGES_REQUESTED** [review]
You're going too far

File: code/datums/station_traits/positive_traits.dm (hunk 13 above)
Comment (Mothblocks) [reply code/datums/station_traits/positive_traits.dm:362]: This fucks over effectively every single antagonist

**Comment by Mothblocks** [comment]
Your "why it's good for the game" is also unchanged when I have previously said it is not adequate since there is no reason to believe anything you have written is good for the game

**Comment by etherware-novice** [comment]
> Your "why it's good for the game" is also unchanged when I have previously said it is not adequate since there is no reason to believe anything you have written is good for the game

by "anything i have written" do you mean in this pr or all of the code ive ever written because if so theres a certain pr i can think of that everyone agreed was a good idea except you.

**Comment by Mothblocks** [comment]
I mean in the "why it's good for the game", which was 6 of the first 7 words of what you just quoted

**Comment by Mothblocks** [comment]
You edited it but this is just exactly the same thing

Why is this good

image

**Comment by Mothblocks** [comment]
We have station traits that make life harder for antagonists, like the deathrattle, but I would argue even those are not as nearly as impactful as "maintenance is near unusable, meaning if you want to do anything, you have to play loud like the game already incentivizes you constantly to do"

**Comment by Mothblocks** [comment]
After talking it out loud I think this is just a stupid idea, I hate how often the game shits on stealth antagonists which leads to the constant problem of antagonists just playing loud, and I think this is in exactly the same camp

**Pull request closed without merging**